mbuild/port.py

```diff
@@ -83,6 +83,46 @@ def _clone(self, clone_of=None, root_container=None):
         newone.used = self.used
         return newone
 
+    def update_separation(self, separation):
+        """
+        Change the distance between a port and its anchor particle
+
+        separation : float, required
+            Distance to shift port along the orientation vector from the anchor
+            particle position. If no anchor is provided, the port will be shifted
+            from the origin.
+        """
+        if self.used:
+            warn("This port is already being used and changing its separation"
+                    " will have no effect on the distance between particles.")
+
+        if self.anchor:
+            self.translate_to(self.anchor.pos)
+        else:
+            self.translate_to((0,0,0))
+
+        self.translate(separation*self.direction)
+
+    def update_orientation(self, orientation):
+        """
+        Change the direction between a port and its anchor particle
+
+        orientation : array-like, shape=(3,), optional, default=[0, 1, 0]
+            Vector along which to orient the port
+        """
+        if self.used:
+            warn("This port is already being used and changing its orientation"
+                    " will have no effect on the direction between particles.")
+
+        orientation = np.asarray(orientation)
```

Comment:
Maybe we should reshape the numpy array `orientation` to enforce it is the correct dimensions?
```suggestion
        orientation = np.asarray(orientation).reshape(3,)
```

I think that is the correct way use the `reshape` method. it might be
```suggestion
        orientation = np.asarray(orientation).reshape((3,))
```